### DISCUSSION TITLE
I want if else conditional in controller function [v5]

### DISCUSSION THREAD

**Discussion by b23618** [adonisjs/core#2987] [Help] [answered]
`const reqData = request.input("lottoticket_group");
      if (request.input("lottoticket_group")) {
        let lottoresult = await LottoryTicketGroup.query()
          .preload("LottoRecord", (LottoRecordQuery) => {
            LottoRecordQuery.orderBy("lotto_name_id", "asc").limit(13);
          })
          .where("id", reqData)
          .orderBy("id", "desc")
          .first();
      }
      let lottoresult = await LottoryTicketGroup.query()
        .preload("LottoRecord", (LottoRecordQuery) => {
          LottoRecordQuery.orderBy("lotto_name_id", "asc").limit(13);
        })
        .orderBy("id", "desc")
        .first();`

I want merge coding if else under in function 

Thank you

**Comment by arthurcvm** [comment, marked as answer]
Hi @menphurk!
`await` and `.orderBy("id", "desc").first()` run the query. You can make this:

```ts
const reqData = request.input("lottoticket_group")

let lottoresult = LottoryTicketGroup.query()
    .preload("LottoRecord", (LottoRecordQuery) => {
        LottoRecordQuery.orderBy("lotto_name_id", "asc").limit(13)
    })

if (reqData) {
    lottoresult.where("id", reqData)
}

const result = await lottoresult.orderBy("id", "desc").first()
```

docs: https://docs.adonisjs.com/guides/models/crud-operations#using-the-query-builder

**Reply by RomainLanz** [reply]
Hey! 👋 

It would be better to use the `if()` method.

```ts
LottoryTicketGroup.query()
  .preload("LottoRecord", (LottoRecordQuery) => {
    LottoRecordQuery.orderBy("lotto_name_id", "asc").groupLimit(13)
  })
   .if(reqData, (query) => {
    query.where("id", reqData)
  })
```
📚  https://docs.adonisjs.com/reference/database/query-builder#if

Also note the usage of `groupLimit`.

📚 https://docs.adonisjs.com/reference/orm/relations/has-many-through#grouplimit
